File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/python_solutions/2952-minimum-number-of-coins-to-be-added.py
class Solution:
    def minimumAddedCoins(self, coins: List[int], target: int) -> int:
        coins.sort()
        res = 0
        current_max = 0
        i = 0
        n = len(coins)

        while current_max < target:
            if i < n and coins[i] <= current_max + 1:
                current_max += coins[i]
                i += 1
            else:
                res += 1
                current_max += current_max + 1

        return res

File: leetcode/AIcoding/5.Qwen-3/cpp_solutions/2952-minimum-number-of-coins-to-be-added.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minimumCoins(vector<int>& coins, int target) {
        sort(coins.begin(), coins.end());
        int res = 0;
        int curr = 1;
        int i = 0;
        while (curr <= target) {
            if (i < coins.size() && coins[i] <= curr) {
                curr += coins[i];
                i++;
            } else {
                curr += curr;
                res++;
            }
        }
        return res;
    }
};